daft/expressions/visitor.py

```diff
@@ -151,3 +153,171 @@ def visit_is_null(self, expr: Expression) -> R:
     def visit_not_null(self, expr: Expression) -> R:
         """Visit an not_null predicate."""
         ...
+
+
+class _PyArrowExpressionVisitor(PredicateVisitor[pc.Expression]):
+    """This visitor does a tree fold into the pyarrow.compute.Expression domain."""
+
+    def visit_col(self, name: str) -> pc.Expression:
+        """Convert the daft column to pc field reference by name."""
+        return pc.field(name)
```

Comment:
Do we need any special handling for nested fields, like in structs? I don't know how Daft represents it, but PyArrow uses tuples: https://arrow.apache.org/docs/python/generated/pyarrow.compute.field.html.

Reply:
We don't have the concept of nested field reference AFAIK. I believe daft only has top-level column reference, then we have the struct expressions to .get() nested fields.